### PR TITLE
[paymentservice] Adds otel logging support

### PR DESCRIPTION
**Pull request by Kimbohlovette**
This PR adds otel instrumentation of the pino library to the payment service

### CODE REVIEW

**Review by julianocosta89 — COMMENTED** [review]
Hello @Kimbohlovette,

I've tested this PR and couldn't see any logs being exported via OTLP.

I do see trace data being added to the logs, but the logs are not being sent to the collector.

**Comment by Kimbohlovette** [comment]
Okay, similar thing as with the accounting service. Any recommendations on what to do?
@puckpuck

File: src/paymentservice/opentelemetry.js [hunk 1]
```diff
@@ -1,41 +1,72 @@
...
+} = require('@opentelemetry/resource-detector-container');
+const { gcpDetector } = require('@opentelemetry/resource-detector-gcp');
+const {
+	envDetector,
+	hostDetector,
+	osDetector,
+	processDetector,
+} = require('@opentelemetry/resources');
+
+const { PinoInstrumentation } = require('@opentelemetry/instrumentation-pino');
```
Comment (puckpuck) [inline src/paymentservice/opentelemetry.js:33]: Can you reformat the require statements to all be 1 line each, similar to how we have this done for other JS/TS files

**Review by puckpuck — CHANGES_REQUESTED** [review]

File: src/paymentservice/opentelemetry.js (hunk 1 above)
Comment (Kimbohlovette) [reply src/paymentservice/opentelemetry.js:33]: Okay

**Comment by Kimbohlovette** [comment]
@puckpuck I've just reformatted the require statements as you requested. Could you take a look?

**Review by austinlparker — CHANGES_REQUESTED** [review]
See comment on #1477

**Comment by Kimbohlovette** [comment]
@puckpuck @austinlparker @julianocosta89 is there a library or a stable `exporter` for `javascript` that can be used to export logs to opentelemetry?

**Comment by julianocosta89** [comment]
I spoke with @Kimbohlovette today and asked him to reach out to the otel-js team and check if the OTel logs are actually usable.
Whenever he gets a reply, he will either adjust this PR to properly export the logs, or close it as not implemented yet.

**Comment by trentm** [comment]
I answered on Slack as well, but for the record here: the OTel JS package `@opentelemetry/instrumentation-pino` does not *yet* support the logs bridge API for sending log records via OTLP. I intend to work on that, but haven't gotten to it yet. (FWIW, the other instrumentation JS logging framework instrumentations -- Bunyan and Winston -- *have* added logs bridge support recently.)

**Comment by Kimbohlovette** [comment]
@julianocosta89 I've done discussed with **trentm** I will use `bunyan` to implement this. So work is in progress

**Comment by Kimbohlovette** [comment]
@trentm @julianocosta89 I have replaced pino with bunyan and done the neccessary instrumentation. 
I would like you to test and check if logs and coming to otel correctly. And I would also like to ask if checking these logs in opentelemetry is something that I can do locally before I push

**Comment by julianocosta89** [comment]
@Kimbohlovette you can build this service locally and then start the demo locally.
Once everything is up and running, you can navigate to grafana:
`localhost:8080/grafana` and check if the logs are arriving in the demo chart.

**Comment by Kimbohlovette** [comment]
> @Kimbohlovette you can build this service locally and then start the demo locally. Once everything is up and running, you can navigate to grafana: `localhost:8080/grafana` and check if the logs are arriving in the demo chart.

Okay I am on it

**Comment by Kimbohlovette** [comment]
@julianocosta89, I've updated the environment variables. I would like you to look at the instrumentation.
Aside from that I looked at the log api which is currently in `experimental` and it looks like something that can work fine if we we want to keep things simple

File: .env [hunk 2]
```diff
@@ -1,5 +1,3 @@
-
```
Comment (julianocosta89) [inline .env:1]: This file has no changes, please remove it from the PR.

File: src/paymentservice/index.js [hunk 3]
```diff
@@ -1,59 +1,68 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
-const grpc = require('@grpc/grpc-js')
```
Comment (julianocosta89) [inline src/paymentservice/index.js:3]: @Kimbohlovette could you keep the same indentation we had before?

It would make our life as reviewers easier

Comment (Kimbohlovette) [reply src/paymentservice/index.js:3]: okay

Comment (Kimbohlovette) [reply src/paymentservice/index.js:3]: okay

**Comment by Kimbohlovette** [comment]
> > @julianocosta89, I've updated the environment variables. I would like you to look at the instrumentation. Aside from that I looked at the log api which is currently in `experimental` and it looks like something that can work fine if we we want to keep things simple
> 
> We would like to have OTLP logs, if it works we will definitely get it merged.

Okay

**Comment by puckpuck** [comment]
Please update your IDE settings to not modify the indentation and line wrap of code files. This will make reviewing the PR much easier.

I ran this branch, and did not see the traces or logs being emitted from the payment service. With a quicker look I noticed you added console exporters for both. I don't think you need any exporter, since the default should be OTLP, which is configured using environment variables.

**Comment by Kimbohlovette** [comment]
> Please update your IDE settings to not modify the indentation and line wrap of code files. This will make reviewing the PR much easier.
> 
> I ran this branch, and did not see the traces or logs being emitted from the payment service. With a quicker look I noticed you added console exporters for both. I don't think you need any exporter, since the default should be OTLP, which is configured using environment variables.

Alright, I've been finding out how to disable that auto formatting on vscode for some time now

**Comment by Kimbohlovette** [comment]
Hello @pichlermarc, I have tried to use `@opentelemetry/sdk-logs` in this service to send logs to open telemetry, Unfortunately I can not see any logs emitted on `grafana`.
Please can you look at my code to see what is not right?

**Comment by Kimbohlovette** [comment]
@trentm  I tried bunyan but no lock too

**Comment by pichlermarc** [comment]
> Hello @pichlermarc, I have tried to use `@opentelemetry/sdk-logs` in this service to send logs to open telemetry, Unfortunately I can not see any logs emitted on `grafana`. Please can you look at my code to see what is not right?

Looking at this PR's diff, it looks like there's no exporter configured other than the console log exporter. Are you using a different set of changes to export to grafana/loki? :thinking:

**Comment by Kimbohlovette** [comment]
> > Hello @pichlermarc, I have tried to use `@opentelemetry/sdk-logs` in this service to send logs to open telemetry, Unfortunately I can not see any logs emitted on `grafana`. Please can you look at my code to see what is not right?
> 
> Looking at this PR's diff, it looks like there's no exporter configured other than the console log exporter. Are you using a different set of changes to export to grafana/loki? 🤔

I don't full understanding on how the logs gets to grafana dashboard but what I understand is that in this exercise open telemetry has already been configured to send logs to grafana (which is one of the servies in this app serving on 8080/grafana).

So the goal of this issue is get the app to send logs to open telemetry and then it will show on grafana under the current service

**Comment by david1542** [comment]
hi guys, is there any news on that? I also want to enable logging in the payment service.

**Pull request closed without merging**